File: front/mobile_apps/leopardo_core/lib/models/payment_document.dart
class PaymentDocument {
  const PaymentDocument({
    required this.id,
    required this.documentType,
    required this.status,
    required this.isDownloadable,
    this.companyId,
    this.employeeId,
    this.payrollRunId,
    this.paySlipId,
    this.salaryAdvanceId,
    this.filename,
    this.mimeType,
    this.sizeBytes,
    this.errorMessage,
    this.generatedAt,
    this.createdAt,
  });

  final int id;
  final String? companyId;
  final int? employeeId;
  final int? payrollRunId;
  final int? paySlipId;
  final int? salaryAdvanceId;
  final String documentType;
  final String status;
  final String? filename;
  final String? mimeType;
  final int? sizeBytes;
  final bool isDownloadable;
  final String? errorMessage;
  final DateTime? generatedAt;
  final DateTime? createdAt;

  factory PaymentDocument.fromJson(Map<String, dynamic> json) {
    return PaymentDocument(
      id: _asInt(json['id']),
      companyId: json['company_id']?.toString(),
      employeeId: _nullableInt(json['employee_id']),
      payrollRunId: _nullableInt(json['payroll_run_id']),
      paySlipId: _nullableInt(json['pay_slip_id']),
      salaryAdvanceId: _nullableInt(json['salary_advance_id']),
      documentType: (json['document_type'] ?? 'payment_receipt').toString(),
      status: (json['status'] ?? 'pending').toString(),
      filename: json['filename']?.toString(),
      mimeType: json['mime_type']?.toString(),
      sizeBytes: _nullableInt(json['size_bytes']),
      isDownloadable: json['is_downloadable'] == true,
      errorMessage: json['error_message']?.toString(),
      generatedAt: _parseDate(json['generated_at']),
      createdAt: _parseDate(json['created_at']),
    );
  }

  bool get isAvailable => status == 'available' && isDownloadable;

  String get typeLabel => switch (documentType) {
    'advance_receipt' => 'Recu avance',
    'payment_slip' => 'Bulletin',
    'payroll_summary' => 'Resume paie',
    'payment_receipt' => 'Recu paiement',
    _ => 'Document paiement',
  };

  String get statusLabel => switch (status) {
    'pending' => 'En attente',
    'generating' => 'Generation',
    'available' => 'Disponible',
    'failed' => 'Erreur',
    _ => status,
  };

  static int _asInt(dynamic value) {
    if (value is int) return value;
    if (value is num) return value.toInt();
    return int.tryParse(value?.toString() ?? '') ?? 0;
  }

  static int? _nullableInt(dynamic value) {
    if (value == null) return null;
    if (value is int) return value;
    if (value is num) return value.toInt();
    return int.tryParse(value.toString());
  }

  static DateTime? _parseDate(dynamic value) {
    if (value == null) return null;
    return DateTime.tryParse(value.toString());
  }
}

File: api/app/Models/PaymentDocument.php
<?php

namespace App\Models;

use App\Traits\BelongsToCompany;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Carbon;

/**
 * @property int $id
 * @property string|null $company_id
 * @property int|null $employee_id
 * @property int|null $payroll_run_id
 * @property int|null $pay_slip_id
 * @property int|null $salary_advance_id
 * @property string $document_type
 * @property string $status
 * @property string $disk
 * @property string|null $path
 * @property string|null $filename
 * @property string $mime_type
 * @property int|null $size_bytes
 * @property string|null $error_message
 * @property array<mixed>|null $metadata
 * @property int|null $requested_by
 * @property Carbon|null $generated_at
 * @property Carbon|null $created_at
 * @property Carbon|null $updated_at
 */
class PaymentDocument extends Model
{
    use BelongsToCompany;

    public const TYPE_PAYMENT_RECEIPT = 'payment_receipt';

    public const TYPE_PAYMENT_SLIP = 'payment_slip';

    public const TYPE_ADVANCE_RECEIPT = 'advance_receipt';

    public const TYPE_PAYROLL_SUMMARY = 'payroll_summary';

    public const STATUS_PENDING = 'pending';

    public const STATUS_GENERATING = 'generating';

    public const STATUS_AVAILABLE = 'available';

    public const STATUS_FAILED = 'failed';

    /** @var list<string> */
    public const TYPES = [
        self::TYPE_PAYMENT_RECEIPT,
        self::TYPE_PAYMENT_SLIP,
        self::TYPE_ADVANCE_RECEIPT,
        self::TYPE_PAYROLL_SUMMARY,
    ];

    protected $fillable = [
        'company_id',
        'employee_id',
        'payroll_run_id',
        'pay_slip_id',
        'salary_advance_id',
        'document_type',
        'status',
        'disk',
        'path',
        'filename',
        'mime_type',
        'size_bytes',
        'error_message',
        'metadata',
        'requested_by',
        'generated_at',
    ];

    protected $casts = [
        'metadata' => 'array',
        'generated_at' => 'datetime',
        'size_bytes' => 'integer',
    ];

    /** @return BelongsTo<Employee, $this> */
    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    /** @return BelongsTo<Employee, $this> */
    public function requester(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'requested_by');
    }

    /** @return BelongsTo<PayrollRun, $this> */
    public function payrollRun(): BelongsTo
    {
        return $this->belongsTo(PayrollRun::class, 'payroll_run_id');
    }

    /** @return BelongsTo<PaySlip, $this> */
    public function paySlip(): BelongsTo
    {
        return $this->belongsTo(PaySlip::class, 'pay_slip_id');
    }

    /** @return BelongsTo<SalaryAdvance, $this> */
    public function salaryAdvance(): BelongsTo
    {
        return $this->belongsTo(SalaryAdvance::class, 'salary_advance_id');
    }

    /**
     * @param  Builder<static>  $query
     * @return Builder<static>
     */
    public function scopeAvailableToEmployee(Builder $query, Employee $employee): Builder
    {
        return $query
            ->where('company_id', $employee->company_id)
            ->where('employee_id', $employee->id);
    }
}
